### ISSUE TITLE
iOS RocksDB Crash

### ISSUE THREAD

**Issue by jerryliurui** [facebook/rocksdb#5591]
> Note: Please use Issues only for bug reports. For questions, discussions, feature requests, etc. post to dev group: https://www.facebook.com/groups/rocksdb.dev

Hi
My iOS  app which used rocksdb to read and write received many crash like attach image.

so please help me.

![屏幕快照 2019-07-18 下午4 59 11](https://user-images.githubusercontent.com/6234801/61444246-9823cd00-a97d-11e9-892f-f12b147abb30.png)

**Comment by jerryliurui** [comment]
enum Code {
    kOk = 0,
    kNotFound = 1,
    kCorruption = 2,
    kNotSupported = 3,
    kInvalidArgument = 4,
    kIOError = 5,
    kMergeInProgress = 6,
    kIncomplete = 7,
    kShutdownInProgress = 8,
    kTimedOut = 9,
    kAborted = 10,
    kBusy = 11,
    kExpired = 12,
    kTryAgain = 13
  };

my errorcode.

Is that possible i have missed：
    kCompactionTooLarge = 14,
    kColumnFamilyDropped = 15,
    kMaxCode

**Comment by siying** [comment]
SyncPoint? Do you build release or debug build?

**Comment by jerryliurui** [comment]
> SyncPoint? Do you build release or debug build?

Just Run: TARGET_OS=IOS make static_lib.

**Comment by siying** [comment]
@jerryliurui yes it is debug build. To build release you need to either put `TARGET_OS=IOSrelease`, or "TARGET_OS=IOS DEBUG_LEVEL=0 make static_lib".

I'm not trying to explain the problem though. It's not expected even with debug build.

**Comment by jerryliurui** [comment]
> @jerryliurui yes it is debug build. To build release you need to either put `TARGET_OS=IOSrelease`, or "TARGET_OS=IOS DEBUG_LEVEL=0 make static_lib".
> 
> I'm not trying to explain the problem though. It's not expected even with debug build.

Thank You！
I will try it！

**Comment by jerryliurui** [comment]
> @jerryliurui yes it is debug build. To build release you need to either put `TARGET_OS=IOSrelease`, or "TARGET_OS=IOS DEBUG_LEVEL=0 make static_lib".
> 
> I'm not trying to explain the problem though. It's not expected even with debug build.

HI
I saw this line in makefile like:
# * DEBUG_LEVEL=0; this is the debug level we use for release. If you're
# running rocksdb in production you most definitely want to compile RocksDB
# with debug level 0. To compile with level 0, run `make shared_lib`,
# `make install-shared`, `make static_lib`, `make install-static` or
# `make install`

So Just Run: TARGET_OS=IOS make static_lib whether is equal TARGET_OS=IOS DEBUG_LEVEL=0 make static_lib?
